specification/syntex/resource-manager/Microsoft.Syntex/preview/2022-09-15-preview/syntex.json

```diff
@@ -0,0 +1,521 @@
+          },
+          {
+            "$ref": "#/parameters/ProcessorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Document processor details.",
+            "schema": {
+              "$ref": "#/definitions/DocumentProcessorResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DocumentProcessor_Get": {
+            "$ref": "./examples/DocumentProcessor_Get.json"
+          }
+        }
+      },
+      "delete": {
+        "description": "Deletes document processor resource for a given name.",
+        "operationId": "DocumentProcessor_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProcessorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Deletes the document processor details."
+          },
+          "204": {
+            "description": "The document processor details are not present."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DocumentProcessor_Delete": {
+            "$ref": "./examples/DocumentProcessor_Delete.json"
+          }
+        }
+      },
+      "put": {
+        "description": "Creates a document processor resource for a given name.",
+        "operationId": "DocumentProcessor_Create",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProcessorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "body",
+            "description": "Document processor details.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DocumentProcessorResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DocumentProcessorResource"
+            }
+          },
```

Comment:
PUT needs to return 201 Created responses for initial create, as well as 200 for updates

Reply:
It seems that returning 201 makes PUT an asynchronous operation, with the linter validation error:
[RPaaS] An async PUT operation must set long running operation options 'x-ms-long-running-operation-options'

At the ARM review board meeting earlier this year we agreed that this RP does not need asynchronous operations.

Reply:
See similar thread above, 201 != Async.